File: backend/src/main/java/com/project/product/service/OrderService.java
package com.project.product.service;

import com.project.product.controller.order.model.OrderCreateModel;
import com.project.product.controller.order.model.OrderItemModel;
import com.project.product.controller.order.model.OrderModel;

import java.util.List;
import java.util.Optional;

public interface OrderService {
    OrderCreateModel save(OrderCreateModel orderCreateModel);
    List<OrderModel> getOrdersForCurrentUser();
    List<OrderModel> getAll();
    Optional<OrderModel> getById(Long id);
    OrderModel update(OrderModel orderModel);
    void delete(Long id);

    List<OrderItemModel> getOrderItemsByOrderId(Long orderId);
}

File: src/app/orders/services/order.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { OrderCreateModel, OrderItemModel, OrderModel } from '../models/order.models';

@Injectable({
  providedIn: 'root'
})
export class OrderService {
  private baseUrl = 'http://localhost:8080/api/orders';

  constructor(private http: HttpClient) {}

  // Sipariş oluşturma
  createOrder(order: OrderCreateModel): Observable<OrderCreateModel> {
    return this.http.post<OrderCreateModel>(`${this.baseUrl}/save`, order);
  }

  // Kullanıcının kendi siparişlerini getirme
  getUserOrders(): Observable<OrderModel[]> {
    return this.http.get<OrderModel[]>(`${this.baseUrl}/list`);
  }

  // Tüm siparişleri getirme (admin ve çalışanlar için)
  getAllOrders(): Observable<OrderModel[]> {
    return this.http.get<OrderModel[]>(`${this.baseUrl}/all`);
  }

  // Sipariş detayını getirme
  getOrderById(id: number): Observable<OrderModel> {
    return this.http.get<OrderModel>(`${this.baseUrl}/get/${id}`);
  }

  // Sipariş kalemlerini getirme
  getOrderItems(orderId: number): Observable<OrderItemModel[]> {
    return this.http.get<OrderItemModel[]>(`${this.baseUrl}/getOrderItemsByOrderId/${orderId}`);
  }

  // Sipariş güncelleme
  updateOrder(order: OrderModel): Observable<OrderModel> {
    return this.http.put<OrderModel>(`${this.baseUrl}/update`, order);
  }

  // Sipariş silme
  deleteOrder(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/delete/${id}`);
  }
} 